crates/pubsub/src/implementors/mosquitto.rs

```diff
@@ -0,0 +1,157 @@
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use async_channel::Receiver;
+use futures::executor::block_on;
+use mosquitto_rs::{Client, Message, QoS};
+use slight_common::BasicState;
+
+#[derive(Clone)]
+pub struct MosquittoImplementor {
+    mqtt: Arc<Mutex<Client>>,
+    host: String,
+    port: i32,
+    subscriptions: Arc<Mutex<Vec<String>>>,
+    subscriber: Arc<Mutex<Option<Receiver<Message>>>>,
+}
+
+impl std::fmt::Debug for MosquittoImplementor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "MosquittoImplementor")
+    }
+}
+
+// Pub+Sub
+impl MosquittoImplementor {
+    pub fn new(slight_state: &BasicState) -> Self {
+        let mqtt = Client::with_auto_id().unwrap();
+        let host = String::from_utf8(
+            slight_runtime_configs::get(
+                &slight_state.secret_store,
+                "MOSQUITTO_HOST",
+                &slight_state.slightfile_path,
+            )
+            .with_context(|| {
+                format!(
+                    "failed to get 'MOSQUITTO_HOST' secret using secret store type: {}",
+                    slight_state.secret_store
+                )
+            })
+            .unwrap(),
+        )
+        .unwrap();
+
+        let port = String::from_utf8(
+            slight_runtime_configs::get(
+                &slight_state.secret_store,
+                "MOSQUITTO_PORT",
+                &slight_state.slightfile_path,
+            )
+            .with_context(|| {
+                format!(
+                    "failed to get 'MOSQUITTO_PORT' secret using secret store type: {}",
+                    slight_state.secret_store
+                )
+            })
+            .unwrap(),
+        )
+        .unwrap()
+        .parse::<i32>()
+        .unwrap();
+
+        Self {
+            mqtt: Arc::new(Mutex::new(mqtt)),
+            host,
+            port,
+            subscriptions: Arc::new(Mutex::new(Vec::new())),
+            subscriber: Arc::new(Mutex::new(None)),
+        }
+        // ^^^ arbitrarily chosing to create with 5 threads
```

Comment:
what is this comment about? I didn't see the mentioned 5 threads

Reply:
oops, that was from the old implementation~ good catch! I'll remove it (:

Reply:
done ✅